haystack/nodes/file_converter/azure.py

```diff
@@ -105,22 +109,57 @@ def convert(self,
                                                                           locale=known_language)
             result = poller.result()
 
+        if self.save_json:
+            with open(str(file_path) + ".json", "w") as json_file:
```

Comment:
On windows systems `Path` is more reliable than str. Since `file_path` is already a `Path` let's keep it that type and add the json suffix to it (something like `with_suffix(".json")` should work)